# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1320890003 28800
#      Wed Nov 09 17:53:23 2011 -0800
# Node ID b4b8e525dee0f2a2ecd8e27aed85148510b7ba39
# Parent  40e32fe44aaae3c4f11f60fbae9da8c13c3234b1
Stop debug_on_error triggering for a new plot (bug #32655)

graphics.cc (remove_all_listeners): Turn off debug_on_error, debug_on_warning
when removing all listeners.  Function is known to call get_property on
properties which may not exist.

diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -2806,20 +2806,24 @@ base_graphics_object::remove_all_listene
   for (octave_map::const_iterator pa = m.begin (); pa != m.end (); pa++)
     {
       // FIXME -- there has to be a better way.  I think we want to
       // ask whether it is OK to delete the listener for the given
       // property.  How can we know in advance that it will be OK?
 
       unwind_protect frame;
 
+      frame.protect_var (error_state);
       frame.protect_var (discard_error_messages);
-      frame.protect_var (error_state);
+      frame.protect_var (Vdebug_on_error);
+      frame.protect_var (Vdebug_on_warning);
 
       discard_error_messages = true;
+      Vdebug_on_error = false;
+      Vdebug_on_warning = false;
 
       property p = get_properties ().get_property (pa->first);
 
       if (! error_state && p.ok ())
         p.delete_listener ();
     }
 }
 
